### PR TITLE
Expose all actuators under the CloudFoundry context

### LINKED ISSUE #1293: Requests to /cloudfoundryapplication possibly return results for /actuator
## Describe the bug
In CF context, only /health and /info are reported as being enabled

Sample response:
```json
{
    "type": "steeltoe",
    "_links": {
        "info": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication/info",
            "templated": false
        },
        "health": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication/health",
            "templated": false
        },
        "self": {
            "href": "https://steeltoe-samples-server.apps.dhaka.cf-app.com/cloudfoundryapplication",
            "templated": false
        }
    }
}
```

## Steps to reproduce 
1. Add Actuators to app
2. Deploy to CF
3. View app in apps manager

## Expected behavior
All endpoints supported on Cloud Foundry should be returned/enabled automatically

## Environment (please complete the following information):
 - Steeltoe Version `main`
 - Platform: Cloud Foundry
 - OS: Linux
 - .NET Version 8.0

### PR DESCRIPTION
**Pull request by TimHess**
## Description

Skip the exposure check when building the list of exposed actuators under `/cloudfoundryapplication`

Fixes #1293 

## Quality checklist

- [ ] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [ ] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs [hunk 1]
```diff
@@ -109,8 +108,8 @@ public async Task Invoke_CloudFoundryDisable_ReturnsExpectedLinks()
         var handler = testContext.GetRequiredService<IActuatorEndpointHandler>();
 
         Links links = await handler.InvokeAsync("http://localhost:5000/foobar", CancellationToken.None);
-        Assert.NotNull(links);
-        Assert.NotNull(links.Entries);
-        Assert.Empty(links.Entries);
+        links.Should().NotBeNull();
+        links.Entries.Should().NotBeNull();
+        links.Entries.Should().BeEmpty();
```
Comment (bart-vmware) [inline src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs:113]:
```suggestion
        links.Entries.Should().BeEmpty();
```

`links.Entries.Should().BeEmpty()` already checks for not null, so `links.Entries.Should().NotBeNull()` is redundant.

File: src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs [hunk 2]
```diff
@@ -78,12 +78,11 @@ public async Task Invoke_HonorsEndpointEnabled_ReturnsExpectedLinks()
...
         Links links = await handler.InvokeAsync("http://localhost:5000/foobar", CancellationToken.None);
-        Assert.NotNull(links);
-        Assert.NotNull(links.Entries);
-        Assert.True(links.Entries.ContainsKey("self"));
-        Assert.Equal("http://localhost:5000/foobar", links.Entries["self"].Href);
-        Assert.False(links.Entries.ContainsKey("info"));
-        Assert.Single(links.Entries);
+        links.Should().NotBeNull();
+        links.Entries.Should().NotBeNull();
+        links.Entries.Should().ContainKey("self");
```
Comment (bart-vmware) [inline src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs:83]:
```suggestion
        links.Entries.Should().ContainKey("self");
```

File: src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs (hunk 1 above)
Comment (bart-vmware) [inline src/Management/test/Endpoint.Test/Actuators/Hypermedia/HypermediaEndpointTest.cs:113]:
```suggestion
        links.Entries.Should().BeEmpty();
```
- The `links` variable is declared as non-nullable, so `links.Should().NotBeNull()` looks redundant.
- `links.Entries.Should().BeEmpty()` already checks for not null, so `links.Entries.Should().NotBeNull()` is redundant.

**Review by bart-vmware — CHANGES_REQUESTED** [review]
While I appreciate the effort to convert xUnit assertions to FluentAssertions-style, it should be done with great care. Please carefully read https://fluentassertions.com/tips/, which lists how to write good assertions with FA. Simply converting existing assertions to FA-style syntax makes it questionable why to use the library in the first place.

When I checked a year ago, there existed two analyzers to help with the conversion and suggest better assertions. But when I tried them, they were buggy or had severe performance impact. I'd like to revisit this as part of #931.

File: src/Management/src/Endpoint/Actuators/Hypermedia/HypermediaService.cs
[the remark this answers is not in the document]
Comment (bart-vmware) [reply src/Management/src/Endpoint/Actuators/Hypermedia/HypermediaService.cs:1]: To get rid of the cognitive complexity message, it's sufficient to extract a separate method:
```c#
private void AddToLinkEntries(string baseUrl, Links links, EndpointOptions endpointOptions)
{
    if (!string.IsNullOrEmpty(endpointOptions.Id))
    {
        if (!links.Entries.ContainsKey(endpointOptions.Id))
        {
            string linkPath = $"{baseUrl.TrimEnd('/')}/{endpointOptions.Path}";
            links.Entries.Add(endpointOptions.Id, new Link(linkPath));
        }
        else
        {
            _logger.LogWarning("Duplicate endpoint ID detected: {DuplicateEndpointId}", endpointOptions.Id);
        }
    }
}
```

[the remark this answers is not in the document]
Comment (TimHess) [reply src/Management/src/Endpoint/Actuators/Hypermedia/HypermediaService.cs:1]: This all sounds good to me, thanks - I'll push the changes

**Comment by TimHess** [comment]
> While I appreciate the effort to convert xUnit assertions to FluentAssertions-style, it should be done with great care. Please carefully read https://fluentassertions.com/tips/, which lists how to write good assertions with FA. Simply converting existing assertions to FA-style syntax makes it questionable why to use the library in the first place.
> 
> When I checked a year ago, there existed two analyzers to help with the conversion and suggest better assertions. But when I tried them, they were buggy or had severe performance impact. I'd like to revisit this as part of #931.

I did try to change most/all of the assertions in a file when I ran into stupid test failure messages (false should be true, etc) so we'd at least be closer to where we want to be here, but admittedly put very little effort into true improvements... I'm sure I've _read_ those tips before, but clearly haven't fully absorbed them, so I've bookmarked the page and will attempt to keep them in mind in the future, thanks.

**Review by bart-vmware — DISMISSED** [review]
LGTM, aside from build errors

**Review by bart-vmware — APPROVED** [review]

**Pull request merged**